src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/validation/ItemGroupsValidator.java

```diff
@@ -1,97 +1,142 @@
 package uk.gov.companieshouse.itemgroupworkflowapi.validation;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import org.apache.commons.lang.StringUtils;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.DeliveryDetails;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.Item;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.ItemCostProductType;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.ItemCosts;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.ItemDescriptionIdentifier;
 import uk.gov.companieshouse.itemgroupworkflowapi.model.ItemGroupData;
+import uk.gov.companieshouse.itemgroupworkflowapi.model.ItemKind;
 import uk.gov.companieshouse.itemgroupworkflowapi.model.Links;
-import uk.gov.companieshouse.itemgroupworkflowapi.service.ItemGroupsService;
 
 import java.util.ArrayList;
 import java.util.List;
 
 @Component
 public class ItemGroupsValidator {
-    private static final String COMPANY_NAME_KEY = "company_name_includes";
-    private static final String COMPANY_NUMBER_KEY = "company_number";
-    private static final String COMPANY_NAME_MISSING_MESSAGE = "missing company_name";
-    private static final String COMPANY_NUMBER_INVALID = "order number invalid";
+    public static final String ORDER_NUMBER_INVALID = "Items Group : order number invalid";
+    public static final String ITEM_GROUP_ITEMS_MISSING = "Items Group : items not provided";
+    public static final String ITEM_GROUP_LINKS_MISSING = "Items Group : links not provided";
+    public static final String LINKS_ORDER_NUMBER_MISSING = "Items Group : link order number not provided";
+    public static final String INVALID_DESCRIPTION_IDENTIFIER = "Items Group : invalid item description identifier : ";
+    public static final String INVALID_ITEM_COST_PRODUCT_TYPE = "Items Group : invalid item cost product type : ";
+    public static final String INVALID_ITEM_KIND_NAME = "Items Group : invalid item kind : ";
+    public static final String COMPANY_NUMBER_MISSING = "Items Group Delivery Details : missing item company number for : ";
+    public static final String COMPANY_NAME_MISSING = "Items Group Delivery Details : missing item company name for : ";
+
+    public List<String> validateCreateItemData(ItemGroupData dto) {
+        final List<String> errors = new ArrayList<>();
 
-    private final ItemGroupsService itemGroupsService;
+        if (orderNumberValid(dto, errors) &&
+            itemsPresent(dto, errors) &&
+            linksPresentWithOrderNumber(dto, errors))
```

Comment:
I think we need to validate the links on the items too. Perhaps in a separate PR/ticket?

Reply:
I'll speak to Frank about getting some placeholder tickets added that we can refine and bring in potentially during mid iteration.

Reply:
`DCAC-247` and `DCAC-248` have been raised and moved to the top of the backlog to `Validate item links in item-group-workflow-api` and `Expand ItemGroupController exception handling`